src/server/scripts/Commands/cs_go.cpp

```diff
@@ -288,34 +291,68 @@ class go_commandscript : public CommandScript
         return true;
     }
 
-    //teleport at coordinates, including Z and orientation
-    static bool HandleGoXYZCommand(ChatHandler* handler, float x, float y, Optional<float> z, Optional<uint32> id, Optional<float> o)
+    /**
+     * @brief Teleports the GM to the specified world coordinates, optionally specifying map ID and orientation.
+     *
+     * @param handler The ChatHandler that is handling the command.
+     * @param inputCoords The coordinates to teleport to in format "x y z [mapId [orientation]]".
+     * @return true The command was successful.
+     * @return false The command was unsuccessful (show error or syntax)
+     */
+    static bool HandleGoXYZCommand(ChatHandler* handler, Tail inputCoords)
     {
-        Player* player = handler->GetSession()->GetPlayer();
-        uint32 mapId = id.value_or(player->GetMapId());
+        std::wstring wInputCoords;
+        if (!Utf8toWStr(inputCoords, wInputCoords))
+        {
+            return false;
+        }
 
-        if (z)
+        // extract float and integer values from the input
+        std::vector<float> locationValues;
+        std::wregex floatRegex(L"(-?\\d+(?:\\.\\d+)?)");
+        std::wsregex_iterator floatRegexIterator(wInputCoords.begin(), wInputCoords.end(), floatRegex);
+        std::wsregex_iterator end;
+        while (floatRegexIterator != end)
         {
-            if (!MapMgr::IsValidMapCoord(mapId, x, y, *z))
+            std::wsmatch match = *floatRegexIterator;
+            std::wstring matchStr = match.str();
+
+            // try to convert the match to a float
+            try
             {
-                handler->PSendSysMessage(LANG_INVALID_TARGET_COORD, x, y, mapId);
-                handler->SetSentErrorMessage(true);
-                return false;
+                locationValues.push_back(std::stof(matchStr));
             }
+            // if the match is not a float, do not add it to the vector
+            catch (std::invalid_argument const&){}
+
+            ++floatRegexIterator;
         }
-        else
+
+        // X and Y are required
+        if (locationValues.size() < 2)
         {
-            if (!MapMgr::IsValidMapCoord(mapId, x, y))
-            {
-                handler->PSendSysMessage(LANG_INVALID_TARGET_COORD, x, y, mapId);
-                handler->SetSentErrorMessage(true);
-                return false;
-            }
-            Map const* map = sMapMgr->CreateBaseMap(mapId);
-            z = std::max(map->GetHeight(x, y, MAX_HEIGHT), map->GetWaterLevel(x, y));
+            return false;
+        }
```

Comment:
There should be an error message if not x and y is provided.

Reply:
The current solution causes the syntax help to be shown. What error would you like me to show instead?

Reply:
handler->PSendSysMessage(LANG_INVALID_TARGET_COORD, x, y, mapId);
            handler->SetSentErrorMessage(true);
            return false;

Reply:
I'm so sorry, I don't understand.

At the point in the function that you're suggesting this, I don't have `x`, `y`, or `mapId` yet. In fact, we know that the user hasn't provided at least 2 values, so `x` or `y` are going to be null. Emitting `LANG_INVALID_TARGET_COORD` here also means that it's impossible to get access to the syntax help.

I _do_ check that the coordinates provided are valid further down, and if not I send the `LANG_INVALID_TARGET_COORD` message with the `x`, `y`, and `mapId` values I was able to parse out.
https://github.com/kjack9/azerothcore-wotlk/blob/ca1435384192fc05331b32da82dea69be486d755/src/server/scripts/Commands/cs_go.cpp#L348-L353

Sorry to be missing this one. Let me know what you're wanting in this situation.

Reply:
Most existing commands in my experience do stick the syntax in your face as an error message when improper syntax is supplied, so personally at least I agree with kjack's current implementation.

Reply:
this would give the syntax too. And keep the same behavior as before.

Reply:
never mind my brain is not functioning correctly